Lib/contextlib.py

```diff
@@ -303,6 +303,31 @@ def __exit__(self, *exc_info):
         self.thing.close()
 
 
+class aclosing(AbstractAsyncContextManager):
+    """Async context manager for safely finalizing an async generator using its ``aclose()`` method.
```

Comment:
Is this description too specific?  `aclosing` can be used with any object having `aclose()`, not necessarily a generator.

Reply:
Updated a little to embrace other cases.